src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Parser/ParserHelpers.cs

```diff
@@ -70,5 +72,23 @@ public static bool TryParseThrowFormatException(out int bytesConsumed)
             value = default;
             return TryParseThrowFormatException(out bytesConsumed);
         }
+
+        //
+        // Enable use of ThrowHelper from TryParse() routines without introducing dozens of non-code-coveraged "value= default; bytesConsumed = 0; return false" boilerplate.
+        //
+        [DoesNotReturn]
+        [StackTraceHidden]
+        public static bool TryParseThrowFormatException<T>(ReadOnlySpan<byte> source, out T value, out int bytesConsumed) where T : struct
```

Comment:
This adds more generic instantiations of the helper. Is that a net win or loss? asm size/quality is more important than perfect code coverage stats.

Reply:
I think this should be ok. There are two generic overloads under consideration:

```cs
public static bool TryParseThrowFormatException<T>(out T value, out int bytesConsumed) where T : struct { }
public static bool TryParseThrowFormatException<T>(ReadOnlySpan<byte> source, out T value, out int bytesConsumed) where T : struct { }
```

The first overload is used only for _T_ = { `DateTime`, `DateTimeOffset`, `decimal`, `TimeSpan` }. The second overload is used only for _T_ = { `bool`, `Guid`, \[integral types\] }. Since no _T_ results in both methods being instantiated I don't believe it will cause bloat.

Somebody should sanity check that statement. :)
@davidwrighton, @EgorBo, do you have thoughts?

Reply:
@GrabYourPitchforks Yes, this shouldn't have much of an effect. I'd be concerned if there was an actual generic code which called these apis, but since they aren't actually called from generic code themselves, there isn't a virality problem here or a meaningful code expansion issue.